ckanext/sitemap/view.py

```diff
@@ -73,38 +57,29 @@ def _create_file(filename, root):
             url = etree.SubElement(root, "url")
             loc = etree.SubElement(url, "loc")
             loc.text = _url
-            _create_language_alternatives(_url, url)
 
-        for pkg in pkgs:
+        for pkg in datasets:
             url = etree.SubElement(root, "url")
             loc = etree.SubElement(url, "loc")
-            pkg_url = tk.url_for(controller="dataset", action="read", id=pkg.name)
+            pkg_url = tk.url_for(controller="dataset", action="read", id=pkg["name"])
             loc.text = tk.config.get("ckan.site_url") + pkg_url
             lastmod = etree.SubElement(url, "lastmod")
-            lastmod.text = pkg.metadata_modified.strftime("%Y-%m-%d")
-            _create_language_alternatives(pkg_url, url)
-            for res in pkg.resources:
+            metadata_modifiedDate = datetime.strptime(pkg["metadata_modified"], "%Y-%m-%dT%H:%M:%S.%f")
+            lastmod.text = metadata_modifiedDate.strftime("%Y-%m-%d")
+            resources = list(pkg["resources"])
+
+            for res in resources:
                 url = etree.SubElement(root, "url")
                 loc = etree.SubElement(url, "loc")
                 loc.text = tk.config.get("ckan.site_url") + tk.url_for(
                     controller="dataset_resource",
                     action="read",
-                    id=pkg.name,
-                    package_type=tk.h.default_package_type(),
-                    resource_id=res.id,
+                    id=pkg["name"],
+                    resource_id=res["id"]
                 )
                 lastmod = etree.SubElement(url, "lastmod")
-                _create_language_alternatives(
-                    tk.url_for(
-                        controller="dataset_resource",
-                        action="read",
-                        id=pkg.name,
-                        package_type=tk.h.default_package_type(),
-                        resource_id=res.id,
-                    ),
-                    url,
-                )
-                lastmod.text = res.created.strftime("%Y-%m-%d")
+                resmodifiedDate = datetime.strptime(res["created"], "%Y-%m-%dT%H:%M:%S.%f")
```

Comment:
For resources it should use `last_modified` and if that doesn't exist the `created` field